src/Orckestra.Composer.Search/Services/SearchViewService.cs

```diff
@@ -99,14 +99,32 @@ public virtual async Task<SearchViewModel> GetSearchViewModelAsync(SearchCriteri
                     viewModel.FacetSettings.SelectedFacets,
                     viewModel.ProductSearchResults.CategoryFacetCounts).ConfigureAwait(false);
 
-                // Json context for Facets
-                viewModel.FacetSettings.Context["CategoryFacetValuesTree"] = viewModel.FacetSettings.CategoryFacetValuesTree;
-                viewModel.FacetSettings.Context["SelectedFacets"] = viewModel.FacetSettings.SelectedFacets;
-                viewModel.FacetSettings.Context["Facets"] = viewModel.ProductSearchResults.Facets.Where(f => !f.FieldName.StartsWith(SearchConfiguration.CategoryFacetFiledNamePrefix));
-                viewModel.FacetSettings.Context["PromotedFacetValues"] = viewModel.ProductSearchResults.PromotedFacetValues;
+
+                var selectedFacetValues = new HashSet<string>(
+                    viewModel.FacetSettings?.CategoryFacetValuesTree?.ChildNodes
+                        ?.SelectMany(GetSelectedFacetValues) ?? Array.Empty<string>());
+
+                if (viewModel.FacetSettings != null)
+                {
```

Comment:
viewModel.FacetSettings can't be null here as a new object is created for it at the top

I think you need to check for 
if(viewModel.FacetSettings.SelectedFacets. != null